Mathlib/Control/Combinators.lean

```diff
@@ -68,17 +69,20 @@ def sequence {m : Type u → Type v} [Monad m] {α : Type u} : List (m α) → m
     return (h' :: t')
 
 /-- Executes a list of monadic actions in sequence, discarding the results. -/
+@[deprecated "Use `_root_.sequence` instead." (since := "2025-04-07")]
 def sequence' {m : Type → Type u} [Monad m] {α : Type} : List (m α) → m Unit
   | [] => return ()
   | h :: t => h *> sequence' t
 
 /-- Executes `t` if `b` is `true`, doing nothing otherwise.
 
 See also `when` and `whenM`. -/
+@[deprecated "No replacement in Mathlib." (since := "2025-04-07")]
 def whenb {m : Type → Type} [Monad m] (b : Bool) (t : m Unit) : m Unit :=
```

Comment:
This one I do use downstream, particularly since it's standard in Haskell: https://hackage.haskell.org/package/base-4.21.0.0/docs/Control-Monad.html#v:when. Since there is `if/then` without `else` in do-notation in core, I don't mind deleting it too much, but there's a decent strong argument that it could stay around for discoverability for other Haskell people...
But imo we should mention the `if/then` replacement here.

Reply:
I'll suggest using `if ... then` in `do` notation.